### DISCUSSION TITLE
Only show notification on specific monitor?

### DISCUSSION THREAD

**Discussion by winteriscariot** [ErikReider/SwayNotificationCenter#462] [Q&A]
I have 3 monitors and like to have my notifications on the middle one, even if my focus is on one of the side panels. mako has an 'output' option; does swaync have anything similar?

Similar question but not identical to #374 as I want *all* notifications pinned to a specific output.

Thanks!

**Comment by Posty2k3** [comment]
I was wondering the same thing. I've done tons of searches, reading through the man pages, etc. This was the first reference I've seen to someone having the same "problem" as me. I'm not a fan of the notifications appearing on the monitor where my cursor is currently and would like to configure all my notifications to display on the same monitor if possible.

I figured I'd at least add my comment here since it doesn't look like anyone else responded.

**Comment by rbmarliere** [comment]
You could try setting `"layer-shell": false` and rely on scripting and using something like `swaymsg "[app_id=swaync] move to output $OUTPUT"`. I have not tested the viability, though. A quick glance at the code points me here but I'm also not sure: https://github.com/ErikReider/SwayNotificationCenter/blob/01eb2221a0c0c8d814604074c0181a0c8844f548/src/swayncDaemon/swayncDaemon.vala#L134

**Comment by rbmarliere** [comment]
Here's what you need

~/.config/swaync/config.json:
```
{
"layer-shell": false`
}
```

~/.config/sway/config:
```
no_focus [app_id=swaync]
set $out2 "your-preferred-output"  # from `swaymsg -t get_outputs`
for_window [app_id=swaync] move to output $out2, border none, move position 0 ppt 0 ppt
```

**Reply by rbmarliere** [reply]
BTW you might also want to switch focus to the control center, but since it would fall under the same rules it would be created without focus, e.g.:

~/.config/sway/config:
```
bindsym $mod+alt+n exec swaync-client -t -sw && sleep .1 && swaymsg [app_id=swaync] focus
```

**Comment by Posty2k3** [comment]
Any ideas on a solution for this within Hyprland? Setting `"layer-shell":false` results in the notification center showing like the screenshot below where it has a bizarre outer window to the panel itself. I'm also not sure if this is the solution I'd necessarily be looking for, since I like being able to open the notification center on all the monitors given the media widget, just prefer having the actual notification popups on a single screen.

![image](https://github.com/user-attachments/assets/78e2c3ea-25b9-4c70-b685-23acce642db9)

**Reply by rbmarliere** [reply]
Hmm sorry I've never used hyprland so no idea.

**Comment by flowthentic** [comment]
It always opens on the focused monitor, therefore you can keep `layer-shell` enabled. But before opening the swaync-client, you need to focus the desired output.

My sway configuration stands like this:
`bindsym $mod+n exec "sh -c 'swaymsg focus output eDP-1; swaync-client -op -sw'"`

**Comment by ErikReider** [comment]
This is fixed in the latest release with the new config options and the swaync-client --change-cc-monitor and --change-noti-monitor commands :)
